### PR TITLE
feat: go proxy code and add build and deploy scripts

### PR DESCRIPTION
**Pull request by tmastrom**
Go proxy code and Dockerfile and `cloudbuild.yaml` for deployment.

### CODE REVIEW

**Comment by hanspagel** [comment]
I’ve added a few minor improvements to the docs and the logging. 💅

**Review by hanspagel — APPROVED** [review]
Let’s goooo!

**Comment by tmastrom** [comment]
@hanspagel Thank you for polishing this one up! 💅

File: examples/proxy-server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Build image
+FROM golang:1.22-alpine as build
+WORKDIR /app
+
+COPY ./main.go ./
+RUN go build main.go
+
+# Build distroless image
+FROM gcr.io/distroless/static-debian11
+ENV PORT=9033
```
Comment (geoffgscott) [inline examples/proxy-server/Dockerfile:10]: Please remove for CB

**Review by geoffgscott — APPROVED** [review]

**Pull request merged**